scout/server/blueprints/dashboard/controllers.py

```diff
@@ -33,125 +37,99 @@ def dashboard_form(request_form=None):
     return form
 
 
-def compose_slice_query(search_type, search_term):
-    """Extract a filter query given a form search term and search type
+def populate_dashboard_data(request: LocalProxy) -> dict:
+    """Collect data to display on the dashboard page"""
 
-    Args:
-        search_type(str): example -> "case:"
-        search_term(str): example -> "17867"
-
-    Returns:
-        slice_query(str): example case:17867
-    """
-    slice_query = None
-    if search_term and search_type:
-        slice_query = "".join([search_type, search_term])
-
-    return slice_query
-
-
-def populate_dashboard_data(request):
-    """Prepate data display object to be returned to the view
-
-    Args:
-        request(flask.rquest): request received by the view
-
-    Returns:
-        data(dict): data to be diplayed in the template
-    """
-    data = {"dashboard_form": dashboard_form(request.form), "search_terms": CASE_SEARCH_TERMS}
+    data = {"dashboard_form": dashboard_form(request.form)}
     if request.method == "GET":
         return data
 
-    allowed_insititutes = [inst[0] for inst in institute_select_choices()]
+    allowed_institutes = [inst[0] for inst in institute_select_choices()]
 
     institute_id = request.form.get(
-        "search_institute", allowed_insititutes[0]
+        "search_institute", allowed_institutes[0]
     )  # GET request has no institute, select the first option of the select
 
-    if institute_id and institute_id not in allowed_insititutes:
+    if institute_id and institute_id not in allowed_institutes:
         flash("Your user is not allowed to visualize this data", "warning")
         redirect(url_for("dashboard.index"))
 
     if institute_id == "All":
         institute_id = None
 
-    search_term = request.form["search_term"].strip() if request.form.get("search_term") else ""
-    slice_query = compose_slice_query(request.form.get("search_type"), search_term)
-    get_dashboard_info(store, data, institute_id, slice_query)
+    get_dashboard_info(adapter=store, data=data, institute_id=institute_id, cases_form=request.form)
     return data
 
 
-def get_dashboard_info(adapter, data={}, institute_id=None, slice_query=None):
-    """Append case data stats to data display object
-    Args:
-        adapter(adapter.MongoAdapter)
-        data(dict): data dictionary to be passed to template
-        institute_id(str): institute id
-        slice_query(str): example case:55888
+def get_dashboard_info(
```

Comment:
As usual, I'm no fan of passing the form to the controller. I think it breaks the abstraction. But we did it here before, and do in many other MVCs so why not - it feels convenient enough.